src/main/java/net/sf/mzmine/modules/peaklistmethods/io/csvexport/CSVExportTask.java

```diff
@@ -75,35 +78,53 @@ public double getFinishedPercentage() {
     }
 
     public String getTaskDescription() {
-        return "Exporting peak list " + peakList + " to " + fileName;
+        return "Exporting peak list(s) " 
+                + Arrays.toString(peakLists) + " to CSV file(s)";
     }
 
     public void run() {
 
         setStatus(TaskStatus.PROCESSING);
 
-        // Open file
-        FileWriter writer;
-        try {
-            writer = new FileWriter(fileName);
-        } catch (Exception e) {
-            setStatus(TaskStatus.ERROR);
-            setErrorMessage("Could not open file " + fileName + " for writing.");
-            return;
+        // Shall export several files?
+        boolean substitute = fileName.getPath().contains(plNamePattern);
+
+        // Total number of rows
+        for (PeakList peakList: peakLists) {
+            totalRows += peakList.getNumberOfRows();
         }
 
-        // Get number of rows
-        totalRows = peakList.getNumberOfRows();
+        // Process peak lists
+        for (PeakList peakList: peakLists) {
 
-        exportPeakList(peakList, writer);
+            // Filename
+            File curFile = fileName;
+            if (substitute) {
+                curFile = new File(fileName.getPath().replaceAll(
+                        Pattern.quote(plNamePattern), peakList.getName()));

```

Comment:
Yep, that would be better. And it makes me think that we should be doing this in the "DirectoryComponent", "FileNameComponent", etc. Because, if I am correct, if we type manually illegal characters in their `JTextField` (rather than using the file chooser), there is no check done there either.